### PR TITLE
Fixed all previous issues

### PR DESCRIPTION
**Pull request by FedG-code**
- Titles are now fixed
- og tags are now in place with just an issue with spaces after punctuation in the description
- the list file now passes through the template, eventually I'll have to generate the file itself from all the blogs being turned into html
- final formatting issue: currently I have a div around the header, yet the div isn't taking the size of the header, this means I'm unable to add a bottom margin to the overall header meaning the text gets cut off at the top of the articles as shown below
titlecutoff

### CODE REVIEW

File: blog/converter.js [hunk 1]
```diff
@@ -1,19 +1,69 @@
...
+    const titleEnd = newHTMLFile.search(jsHeaderEnd);
+    var titleFull = newHTMLFile.substring(titleStart, titleEnd);
+    //remove the header formatting
+    titleFull = titleFull.replace(jsHeaderStart, '');
+    titleFull = titleFull.replace(jsHeaderEnd, '');
+    //while this has removed some of the issue there's an id tag attached to the title
+    const titleIdTag = titleFull.search('">');
+    titleFull = titleFull.substring(titleIdTag);
+    titleFull = titleFull.replace('">', '');
+    console.log(titleFull);
```
Comment (robinglen) [inline blog/converter.js]: console

File: blog/converter.js [hunk 2]
```diff
@@ -1,19 +1,69 @@
...
-   //write the html file with its new name
-   writeFileSync(file.replace('.md', '.html'), newHTMLFile);
-});
\ No newline at end of file
+    //Find and remove the pesky invisible character that messes the first title formatting
+    const readableFile = (fileContent.replace('﻿#', '#'));
+    //pass the file through the converter
+    const newHTMLFile = convertToHTML.makeHtml(readableFile);
+    //separate title from body
+    const jsHeaderStart = '<h1';
```
Comment (robinglen) [inline blog/converter.js]: all this method for finding tags could be much cleaner now youre using cheerio

File: blog/converter.js [hunk 3]
```diff
@@ -1,19 +1,69 @@
...
+    //remove the header formatting
+    titleFull = titleFull.replace(jsHeaderStart, '');
+    titleFull = titleFull.replace(jsHeaderEnd, '');
+    //while this has removed some of the issue there's an id tag attached to the title
+    const titleIdTag = titleFull.search('">');
+    titleFull = titleFull.substring(titleIdTag);
+    titleFull = titleFull.replace('">', '');
+    console.log(titleFull);
+    //find the first image in the file
+    const firstImageLocationStart = newHTMLFile.indexOf('<img src=');
```
Comment (robinglen) [inline blog/converter.js]: Same here, cheerio could be much cleaner way of going this

File: blog/converter.js [hunk 4]
```diff
@@ -1,19 +1,69 @@
...
+    const firstImageLocationEnd = firstImage.indexOf('"');
+    firstImage = firstImage.substring(0, firstImageLocationEnd);
+    //now we have to retrieve the text without the html tags for our description
+    const $ = cheerio.load(newHTMLFile);
+    //take the raw text in the paragraphs
+    var rawDescription = $('p').text();
+    //take the first 100 words
+    rawDescription = rawDescription.split(' ').slice(0, 100).join(' ');
+    //since we have taken from multiple paragraphs there's instances of fullstops without a space after, let's fix that
+    rawDescription = rawDescription.replace('.', '. ');
```
Comment (robinglen) [inline blog/converter.js:32]: I wonder if there is something that could clean up the code here better, like a HTML formatter or copy formatter... but we do it for the description why not the whole article?

File: blog/microservices_solving-a-problem-like-routine_2020-update.html [hunk 5]
```diff
@@ -1,4 +1,67 @@
-<h1 id="microservicessolvingaproblemlikerouting2020update">Microservices: Solving a problem like routing — 2020 update</h1>
+<!DOCTYPE html>
+<html lang="en">
+      <head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <title>Microservices: Solving a problem like routing — 2020 update</title>
```
Comment (robinglen) [inline blog/microservices_solving-a-problem-like-routine_2020-update.html:6]: This list has meta data from one article?

File: blog/template.hbs [hunk 6]
```diff
@@ -0,0 +1,67 @@
+<!DOCTYPE html>
```
Comment (robinglen) [inline blog/template.hbs:1]: If we are moving this into a .css we might want to consider doing this to the homepage and sharing

File: blog/template.hbs [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+<script>
+  window.dataLayer = window.dataLayer || [];
+  function gtag() {
+    dataLayer.push(arguments);
+  }
+  gtag("js", new Date());
+  gtag("config", "G-BGXW90V3SE");
+</script>
+<meta property="og:title" content="{{title}}" />
```
Comment (robinglen) [inline blog/template.hbs:23]: put these with the meta tags

File: blog/template.hbs [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+<meta property="og:description" content="{{description}}" />
+<meta property="og:image" content="{{image}}" />
+</head>
+
+<body>
+
+  <script>
+    // When the user scrolls the page, execute myFunction 
+    window.onscroll = function() {myFunction()};
```
Comment (robinglen) [inline blog/template.hbs:32]: myFunction and using vars inside it could be cleaned up

File: blog/template.hbs [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+      var height = document.documentElement.scrollHeight - document.documentElement.clientHeight;
+      var scrolled = (winScroll / height) * 100;
+      document.getElementById("myBar").style.width = scrolled + "%";
+    }
+  </script>
+
+<div class="page-banner">
+  <nav>
+    
+    <a href="templatedlist.html">
```
Comment (robinglen) [inline blog/template.hbs]: lets use `-` to separate words

File: blog/template.hbs [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+      <img id="index-image" src="../img/gullwing-logo.png" alt ="Gullwing Logo">
```
Comment (robinglen) [inline blog/template.hbs:46]: could you think of a better icon here?

File: blog/template.hbs [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+    </a>
+    
+    <a href="https://www.gullwing.io/">
+      <img id="header-image" src="../img/gullwing-logo.png" alt ="Gullwing Logo">
+    </a>  
+  </nav>
+
+  <div class="header">
+    <div class="progress-container">
+      <div class="progress-bar" id="myBar"></div>
```
Comment (robinglen) [inline blog/template.hbs]: `myBar` could be better named

File: blog/templatedlist.html [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+  gtag("config", "G-BGXW90V3SE");
+</script>
+<meta property="og:title" content="Article List" />
+<meta property="og:description" content="" />
+<meta property="og:image" content="" />
+</head>
+
+<body>
+
+  <script>
```
Comment (robinglen) [inline blog/templatedlist.html]: Do we need this here?

File: blog/templatedlist.html [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+
+<script>
+  window.dataLayer = window.dataLayer || [];
+  function gtag() {
+    dataLayer.push(arguments);
+  }
+  gtag("js", new Date());
+  gtag("config", "G-BGXW90V3SE");
+</script>
+<meta property="og:title" content="Article List" />
```
Comment (robinglen) [inline blog/templatedlist.html]: these tags?

File: blog/microservices_solving-a-problem-like-routine_2020-update.html (hunk 5 above)
Comment (FedG-code) [reply blog/microservices_solving-a-problem-like-routine_2020-update.html:6]: I'm not sure what you're asking here

Comment (robinglen) [reply blog/microservices_solving-a-problem-like-routine_2020-update.html:6]: you can actually ignore this one now

**Pull request closed without merging**